src/util/mw-api.ts

```diff
@@ -249,14 +249,19 @@ export function normalizeMwResponse(response: MwApiQueryResponse): QueryMwRet {
     }
     const articleId = String(id).replace(/ /g, '_')
     if (page.redirects) {
-      page.redirects = page.redirects.map((redirect) => {
-        // The API returns the redirect title (!?), we fake the
-        // redirectId by putting the underscore. That way we
-        // secure the URL rewritting works fine.
-        redirect.title = String(redirect.title).replace(/ /g, '_')
-
-        return redirect
-      })
+      page.redirects = page.redirects
+        // remove redirects in User namespace
+        .filter((redirect) => redirect.ns !== 2)
+        // keep only redirects from special namespaces, from subject (not talk) namespaces and from expressly requested namespaces
+        .filter((redirect) => redirect.ns < 0 || redirect.ns % 2 === 0 || MediaWiki.namespacesToMirror.some((ns) => MediaWiki.namespaces[ns].num === redirect.ns))
```

Comment:
Why are we keeping redirects from special namespaces?

Reply:
Special pages don't show up in prop=redirects

Also the only special pages that show up in the api as redirects are user based. 
- Special:MyLanguage is used a lot for links on wikis with Extension:Translate, but as mentioned won't show up here.
- Special:MyPage / Special:MyTalk won't be used a lot, but can reveal the scraping user/ip. However they also won't show up here, only when directly querying or parsing as title.

Reply:
Let's remove special namespaces then, thank you!